### PR TITLE
updated pip install for freebsd

### PR DESCRIPTION
**Pull request by r-dailey**
Updated the install process for Freebsd to enable more tests

### CODE REVIEW

File: lisa/tools/python.py [hunk 1]
```diff
@@ -52,8 +52,12 @@ def dependencies(self) -> List[Type[Tool]]:
 
     def _install(self) -> bool:
         package_name = "python3-pip"
+        posix_os: Posix = cast(Posix, self.node.os)
```
Comment (squirrelsc) [inline lisa/tools/python.py]: Why does it needs to cast? the self.node.os should be able to used for freebsd too.

Comment (r-dailey) [reply lisa/tools/python.py]: Thats a good question. I don't know why. I just followed the format that I saw in a different install function that already had different install paths for different distros. I will look into it and remove the cast if it is not needed.

Comment (r-dailey) [reply lisa/tools/python.py]: Updated this to use the self.node.os and updated the pip name for the tool specific to freebsd to enable it to detect the install

File: lisa/tools/python.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 if TYPE_CHECKING:
     from lisa.node import Node
 
-from lisa.operating_system import Posix
+from lisa.operating_system import Posix, BSD
```
Comment (LiliDeng) [inline lisa/tools/python.py]: `from lisa.operating_system import BSD, Posix`

File: lisa/tools/python.py [hunk 3]
```diff
@@ -53,7 +55,10 @@ def dependencies(self) -> List[Type[Tool]]:
     def _install(self) -> bool:
         package_name = "python3-pip"
         assert isinstance(self.node.os, Posix)
-        self.node.os.install_packages(package_name)
+        if isinstance(self.node.os, BSD):
+            self.node.os.install_packages("py39-pip")
+        else:
+            self.node.os.install_packages(package_name)
```
Comment (LiliDeng) [inline lisa/tools/python.py]:
```
if isinstance(self.node.os, BSD):
    package_name = "py39-pip"
self.node.os.install_packages(package_name)
```

File: lisa/tools/python.py (hunk 2 above)
Comment (r-dailey) [reply lisa/tools/python.py]: Reordered the import statements.

File: lisa/tools/python.py (hunk 3 above)
Comment (r-dailey) [reply lisa/tools/python.py]: updated the if statement to minimize the code size.

File: lisa/tools/python.py [hunk 4]
```diff
@@ -52,6 +54,8 @@ def dependencies(self) -> List[Type[Tool]]:
 
     def _install(self) -> bool:
         package_name = "python3-pip"
+        if isinstance(self.node.os, BSD):
+            package_name = "py39-pip"
```
Comment (squirrelsc) [inline lisa/tools/python.py]: Is there a version for all py3? py39 may have compatibility issue with newer python versions like 3.10, 3.11.

Comment (r-dailey) [reply lisa/tools/python.py]: This is the most recent version of pip I saw available through the pkg utility. I will look into whether or not there is a generic one.

**Review by squirrelsc — DISMISSED** [review]

**Comment by squirrelsc** [comment]
@LiliDeng LGTM

File: lisa/tools/python.py (hunk 4 above)
Comment (r-dailey) [reply lisa/tools/python.py]: Changed it to installing devel/py-pip so that it is not version dependent.

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**